helloworld-impl/src/main/java/sample/helloworld/impl/TestImpl.java

```diff
@@ -14,26 +14,18 @@
 
 public class TestImpl implements Test {
 
-  //#inject-service
-  private final HelloService helloService;
-
-  @Inject public TestImpl(HelloService helloService) {
-    this.helloService = helloService;
-  }
-  //#inject-service
-
-  public ServiceCall<NotUsed, NotUsed> audit() {
-    //#subscribe-to-topic
-    helloService.greetingsTopic().subscribe() // <-- you get back a Subscriber instance
-        .atLeastOnce(
-            Flow.fromFunction(this::doSomethingWithTheMessage));
-    //#subscribe-to-topic
-    return name -> completedFuture(NotUsed.getInstance());
-  }
-
-  private Done doSomethingWithTheMessage(GreetingMessage message) {
-    System.out.println("Message: " + message);
-    return Done.getInstance();
-  }
+    private final HelloService helloService;
+
+    @Inject
+    public TestImpl(HelloService helloService) {
+        this.helloService = helloService;
+        helloService.greetingsTopic().subscribe()
+                .atLeastOnce(Flow.fromFunction(this::doSomethingWithTheMessage));
+    }
+
+    private Done doSomethingWithTheMessage(GreetingMessage message) {
+        System.out.println("Message: " + message);
+        return Done.getInstance();
+    }
 
```

Comment:
Hi @himani1 , the message consumption happens on `doSomethingWithMessage`. That is, `ServiceFoo` publishes data on the broker and then some other class gets a  `ServiceFoo` client injected (using Guice) and subscribes to it.

In your code sample, the code injection and consumption happen in `HelloService` and the processing of the message is in `doSomethingWithMessage`. That method is a side-effecting function that may convert, store, forward, etc... each item received in an idempotent fashion since the subscription is made in `atLeastOnce` semantics meaning an item may be received twice.